### PR TITLE
Add `input_mappings` and `output_mappings` attributes

### PR DESCRIPTION
**Pull request by gabrielmbmb**
## Description

This PR adds two new arguments:

- `input_mappings` argument to all the kind of steps allowing to map columns/keys from the data used as input for the step to the name of the `inputs` required by it.

  ```python
  class GenerateResponse(Step):
      @property
      def inputs(self) -> List[str]:
          return ["instruction"]
  
      @property
      def outputs(self) -> List[str]:
          return ["response"]
  
      def process(self, inputs: StepInput) -> StepOutput:
          for input in inputs:
              input["response"] = "-------> " + input["instruction"]
          yield inputs
  
  step = GenerateResponse(name="generate_response", input_mappings={"instruction": "prompt})
  # The `process_applying_mappings` will replace all the keys `prompt` by `instruction`
  step.process_applying_mappings([{"prompt": "Is `distilabel` awesome?"}])
  ```

- `output_mappings` argument for the `Step` and `GlobalStep` allowing to rename the `outputs` of the step:

  ```python
  class GenerateResponse(Step):
      @property
      def inputs(self) -> List[str]:
          return ["instruction"]
  
      @property
      def outputs(self) -> List[str]:
          return ["response"]
  
      def process(self, inputs: StepInput) -> StepOutput:
          for input in inputs:
              input["response"] = "-------> " + input["instruction"]
          yield inputs
  
  step = GenerateResponse(name="generate_response", output_mappings={"response": "generation})
  # The `process_applying_mappings` will replace all the output keys `response` by `generation`
  step.process_applying_mappings([{"instruction": "Is `distilabel` awesome?"}])
  ```

### CODE REVIEW

**Review by alvarobartt — COMMENTED** [review]
Hi here! LGTM overall, just to take into account that the DAG validation will need to use the mappings if there, so maybe a `get_inputs` function needs to be implemented. Besides that I would rename `outputs_mappings` to `output_mappings` to be consistent with `input_mappings` and to not add the plural to both words which IMO is confusing. Additionally I would also like to see what would the user workflow be for a step that has a predefined set of inputs already.

**Comment by alvarobartt** [comment]
> Additionally I would also like to see what would the user workflow be for a step that has a predefined set of inputs already.

Would this be an user worflow?

```python
from ... import CustomTask

print(CustomTask.inputs)
# ["question", "options"]
print(CustomTask.outputs)
# ["answer"]

input_mappings = {"question": "Q", "options": "O"}
output_mappings = {"answer": "A"}
```

i.e. to be able to explore the defaults before applying the mappings? Also the mappings should be disabled when there are no inputs / outputs right? Are we also handling that?

File: src/distilabel/pipeline/step/base.py [hunk 1]
```diff
@@ -84,6 +84,7 @@ def process(self, input: *StepInput) -> List[Dict[str, Any]]:
     pipeline: Annotated[
         Union[BasePipeline, None], Field(exclude=True, repr=False)
     ] = None
+    outputs_mappings: Dict[str, str] = {}
```
[the remark this answers is not in the document]
Comment (gabrielmbmb) [reply src/distilabel/pipeline/step/base.py:87]: yes, I doubted with this one, I guess `output_mappings` it's better

File: src/distilabel/pipeline/step/base.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 from distilabel.pipeline.base import BasePipeline, _GlobalPipelineManager
 from distilabel.pipeline.logging import get_logger
 from distilabel.pipeline.serialization import _Serializable
-from distilabel.pipeline.step.typing import GeneratorStepOutput, StepOutput
+from distilabel.pipeline.step.typing import GeneratorStepOutput, StepInput, StepOutput
```
Comment (alvarobartt) [inline src/distilabel/pipeline/step/base.py:27]: Maybe we could move these imports under the `TYPE_CHECKING`

**Review by alvarobartt — APPROVED** [review]

**Pull request merged**